### PR TITLE
Include actions in the list of extensions.

### PR DESCRIPTION
**Pull request by vivek**

### CODE REVIEW

File: src/main/java/org/jenkinsci/extension_indexer/Extension.java [hunk 1]
```diff
@@ -44,31 +45,52 @@
...
 
+    private final List<String> jellyFiles = new ArrayList<String>();
+
     /**
      * {@link Trees} object for {@link #javac}
      */
     public final Trees trees;
 
-    Extension(MavenArtifact artifact, JavacTask javac, Trees trees, TypeElement implementation, TreePath implPath, TypeElement extensionPoint) {
+    Extension(MavenArtifact artifact, JavacTask javac, Trees trees, TypeElement implementation, TreePath implPath, TypeElement extensionPoint,TypeElement action) {
```
Comment (michaelneale) [inline src/main/java/org/jenkinsci/extension_indexer/Extension.java]: :ant: missing space at end

**Comment by michaelneale** [comment]
:bee:

**Comment by vivek** [comment]
@kohsuke needs your review, I have made changes to capture actions and views as discussed yesterday. Summary of changes are:

* Action is refactored in to separate class, both Extension and Action subclasses common base class
* Fixes in ExtensionPointsExtractor to capture Action outside of interface recursion
* Views are aggregated outside of extension recursion
* viewScore per artifact is (total_number_of_views/ total_number_of_extensions_and_action)
* views JSON element is array of key'd map 
`{"views":[{"name":"index", "source":"hudson/model/View/index..jelly"}]}`

Example json:
- extension object https://gist.github.com/vivek/18a4611fb57fbc8837be5b0fb6df180a
- sample artifact: https://gist.github.com/vivek/1070d496506cfd07a4e96218a29fab11

**Comment by kohsuke** [comment]
I think you forgot to commit some files, like `BaseClass.java`

**Comment by vivek** [comment]
Oops, pushed them.

**Pull request merged**